azure-iot-device/tests/common/pipeline/test_pipeline_stages_mqtt.py

```diff
@@ -258,30 +278,28 @@ def transport_function_throws_base_exception(params, stage, mocker, fake_base_ex
 
 @pytest.mark.parametrize("params", connection_ops + pubsub_ops)
 @pytest.mark.describe(
-    "MQTTTransportStage - .run_op() -- called with op that maps directly to protocol client library calls"
+    "MQTTTransportStage - .run_op() -- called with op that maps directly to transport calls"
```

Comment:
>"MQTTTransportStage - .run_op() -- called with op that maps directly to transport calls" [](start = 4, length = 88)

Taking this approach of parametrizing the ops limits us to having the same tests for all ops - which prevents us from being able to test the connect/reconnect/disconnect specific "only one op active at a time, doing another one cancels the pre-existing one" requirement.